zhaquirks/tuya/TS011F_plug.py

```diff
@@ -0,0 +1,112 @@
+"""TS011F plug."""
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import Basic, GreenPowerProxy, Groups, Identify, OnOff, Ota, Scenes, Time
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.smartenergy import Metering
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODEL,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+class TuyaZBMetering(CustomCluster, Metering):
+    """TuyaZBMetering cluster implementation. This implementation divides the KwH for tuya devices"""
+
+    cluster_id = Metering.cluster_id
+    current_summ_delivered = 0x0000
+
+    def _update_attribute(self, attrid, value):
+        if attrid == self.current_summ_delivered:
+            value = value / 100
+        super()._update_attribute(attrid, value)
+
+class TuyaZBElectricalMeasurement(CustomCluster, ElectricalMeasurement):
+    """TuyaZBElectricalMeasurement cluster implementation. This implementation divides the Current for tuya devices"""
+
+    cluster_id = ElectricalMeasurement.cluster_id
+    rms_current = 0x0508
+
+    def _update_attribute(self, attrid, value):
+        if attrid == self.rms_current:
+            value = value / 1000
+        super()._update_attribute(attrid, value)        
+
+class PowerOnState(t.enum8):
+    """Tuya power on state enum."""
+
+    Off = 0x00
+    On = 0x01
+    LastState = 0x02
+
+
+class OnOffRestorePowerCluster(CustomCluster, OnOff):
+    """Tuya on off cluster with restore state."""
+
+    attributes = OnOff.attributes.copy()
+    attributes.update({0x8002: ("power_on_state", PowerOnState)})
```

Comment:
```suggestion
    manufacturer_attributes = {0x8002: ("power_on_state", PowerOnState)}
```

Reply:
Not working:
PowerOnState Value is always None
But I know matt has a PR for implementing this in INIT

Reply:
Then its very likely not working for the moment then i have doing the same change in the tuya INIT.

I can rolling it back for getting it working if its OK for Adminiuga in the PR for it.

Reply:
PS the devs is in time zone US so they is having there night sleep for the moment if not doing dream coding ;-))

Reply:
Using TuyaZBOnOffRestorePowerCluster from INIT: PR1056

Reply:
Only need getting it merged so you is not getting errors in the tests (i have it in the old plug.py then deleting the original code).

Reply:
You need adding some `.` in the end of your new comments and running black for fixing some formatting that is not OK (then looking on the last test done).

Reply:
black for formatting, I thought I did :)
where do I need to add ```.``` ?

Reply:
Line 29 and 37.
Its some missing blank lines and extra spaces but that is black fixing that also some to long strings that is being broken in up OK by black (or its being shorted then deleting the power restore).

Reply:
the last test run: https://github.com/zigpy/zha-device-handlers/pull/1057/checks?check_run_id=3785501424 look in `check flake8`

Reply:
I don't use this part in the code anymore:
```
attributes = OnOff.attributes.copy()
attributes.update({0x8002: ("power_on_state", PowerOnState)})
```

and a lot has changed: 
Need a new test.

Reply:
Hi Matt, Did I forgot something? or how to go on from here?

Reply:
Still line 29 and 37 need being ended with one  `.`  (like line 1 and 45 is done) that black is not fixing the rest is black fixing then you running it.
You is also getting one (or more) error for line 25 then the power on state is not committed but its OK then the PR is merged.

Reply:
Looks like all lines is OK :-)))

Hope the other test is not finding any bad things and that the devs can rerunning the test and merger it if its OK :-)))